service/src/test/java/bio/terra/workspace/db/CronjobDaoTest.java

```diff
@@ -0,0 +1,56 @@
+package bio.terra.workspace.db;
+
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import bio.terra.workspace.common.BaseUnitTest;
+import java.time.Duration;
+import java.util.concurrent.TimeUnit;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+
+public class CronjobDaoTest extends BaseUnitTest {
+  @Autowired CronjobDao cronjobDao;
+
+  @Test
+  public void successfullyClaimJob() throws InterruptedException {
+    String jobName = "TEST_JOB";
+    // Guarantee the job is claimed at the start of this test.
+    assertTrue(cronjobDao.claimJob(jobName, Duration.ZERO));
+    TimeUnit.SECONDS.sleep(5);
+    // Check it's been at least 4 seconds to avoid potential flakiness
+    assertTrue(cronjobDao.claimJob(jobName, Duration.ofSeconds(4)));
+  }
+
+  @Test
+  public void failedClaimsDoNotWrite() throws InterruptedException {
+    String jobName = "TEST_JOB";
+    // Guarantee the job is claimed at the start of this test.
+    assertTrue(cronjobDao.claimJob(jobName, Duration.ZERO));
```

Comment:
Yep. This is mostly set up for the next claim attempt (which intentionally fails) - unless we claim the job right at the start of the test, we don't know how long it's been since the job was last run.